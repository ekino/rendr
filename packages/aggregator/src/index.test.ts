import { createAggregator } from "./index";
import { RequestCtx, Page } from "@ekino/rendr-core";

function createContext(data: {} = {}): RequestCtx {
  const ctx: RequestCtx = {
    // @ts-ignore
    req: jest.fn(),
    // @ts-ignore
    res: jest.fn(),
    pathname: "/",
    params: {},
    query: "/",
    asPath: ""
  };

  return {
    ...ctx,
    ...data
  };
}

describe("test inmemory code", () => {
  it("test createAggregator - empty init", async () => {
    const ctx = createContext();
    const aggregator = createAggregator({});

    const page = await aggregator(new Page(), ctx);

    expect(page).not.toBeNull();
    expect(page.statusCode).toBe(200);
  });

  it("test createAggregator - with handlers", async () => {
    const ctx = createContext();
    const aggregator = createAggregator({
      handler: block => {
        block.settings.test = "Salut";

        return Promise.resolve(block);
      }
    });

    let page = new Page();

    page.blocks.push({
      order: 0,
      container: "body",
      type: "handler",
      settings: {}
    });

    page = await aggregator(page, ctx);

    expect(page).toMatchSnapshot();
  });
});
