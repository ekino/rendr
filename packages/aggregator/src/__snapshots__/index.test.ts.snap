// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test inmemory code test createAggregator - with handlers 1`] = `
Page {
  "blocks": Array [
    Object {
      "container": "body",
      "order": 0,
      "settings": Object {
        "test": "Salut",
      },
      "type": "handler",
    },
  ],
  "cache": Object {
    "ttl": 0,
  },
  "head": Object {
    "defaultTitle": "-",
    "htmlAttributes": Object {},
    "link": "",
    "meta": Array [],
    "title": "-",
    "titleTemplate": "%s",
  },
  "id": "",
  "path": "",
  "settings": Object {},
  "statusCode": 200,
  "template": "default",
  "type": "document",
}
`;
